app/producers/application_producer.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+# Common Kafka producer client
+# https://www.rubydoc.info/gems/ruby-kafka/Kafka/Producer
+class ApplicationProducer < Kafka::Client
+  BROKERS = [ENV['KAFKAMQ']].compact.freeze
+  CLIENT_ID = 'compliance-payload-tracker-producer'
+  SERVICE = 'compliance'
+  # Define TOPIC in the inherited class.
+  # Example:
+  #   TOPIC = 'platform.payload-status'
+
+  class << self
+    private
+
+    def deliver_message(msg)
+      msg = msg.merge(
+        date: DateTime.now.iso8601,
+        service: self::SERVICE,
```

Comment:
No need for the `self::` here. `TOPIC` below won't require it either once it's updated to be a method.

Reply:
`self::` dropped.